src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -1174,6 +1215,15 @@ void ThrowIfNotPeekLockMode()
             }
         }
 
+        void ThrowIfSessionLockLost()
+        {
+            if (this.LinkError != null &&
+                this.LinkError.Condition.Equals(AmqpClientConstants.SessionLockLostError))
+            {
```

Comment:
what if the condition is something else? We still have got a `link.Closed` event. Should this be `if(this.LinkError != null)`?